pandas/core/frame.py

```diff
@@ -2639,6 +2639,194 @@ def to_html(
+            string.
+        index : bool, default True
+            Whether to include index in XML document.
+        root_name : str, default 'data'
+            The name of root element in XML document.
+        row_name : str, default 'row'
+            The name of row element in XML document.
+        na_rep : str, optional
+            Missing data representation.
+        attr_cols : list-like, optional
+            List of columns to write as attributes in row element.
+            Hierarchical columns will be flattened with underscore
+            delimiting the different levels.
+        elem_cols : list-like, optional
+            List of columns to write as children in row element. By default,
+            all columns output as children of row element. Hierarchical
+            columns will be flattened with underscore delimiting the
+            different levels.
+        namespaces : dict, optional
+            All namespaces to be defined in root element. Keys of dict
+            should be prefix names and values of dict corresponding URIs.
+            Default namespaces should be given empty string key. For
+            example, ::
+
+                namespaces = {"": "https://example.com"}
+
+        prefix : str, optional
+            Namespace prefix to be used for every element and/or attribute
+            in document. This should be one of the keys in ``namespaces``
+            dict.
+        encoding : str, default 'utf-8'
+            Encoding of the resulting document.
+        xml_declaration : bool, default True
+            Whether to include the XML declaration at start of document.
+        pretty_print : bool, default True
+            Whether output should be pretty printed with indentation and
+            line breaks.
+        parser : {'lxml','etree'}, default 'lxml'
+            Parser module to use for building of tree. Only 'lxml' and
+            'etree' are supported. With 'lxml', the ability to use XSLT
+            stylesheet is supported.
+        stylesheet : str, path object or file-like object, optional
+            A URL, file-like object, or a raw string containing an XSLT
+            script used to transform the raw XML output. Script should use
+            layout of elements and attributes from original output. This
+            argument requires ``lxml`` to be installed. Only XSLT 1.0
+            scripts and not later versions is currently supported.
+        compression : {{'infer', 'gzip', 'bz2', 'zip', 'xz', None}}, default 'infer'
+            For on-the-fly decompression of on-disk data. If 'infer', then use
+            gzip, bz2, zip or xz if path_or_buffer is a string ending in
+            '.gz', '.bz2', '.zip', or 'xz', respectively, and no decompression
+            otherwise. If using 'zip', the ZIP file must contain only one data
+            file to be read in. Set to None for no decompression.
+        storage_options : dict, optional
```

Comment:
If you want you can re-use the shared doc sting from `pandas/core/shared_docs.py` for storage_options

Reply:
I tried using the decorator, `@doc(storage_options=generic._shared_docs["storage_options"])` inside frame.py for `to_xml` and errs `KeyError '"'"`. See traceback below. I even tried many, many ways to use the decorator in pandas.io.xml for `read_xml`. But it always errs on `KeyError '"doc"'` (pointing to pandas.util._decorators.py). What am I missing? I `fetch/merge upstream` and rebuilt C-extensions with `python setup.py build_ext -j 4` to no avail.

```
Traceback (most recent call last):
  File "sandbox.py", line 5, in <module>
    import pandas as pd
  File "/home/pandas-parfaitg/pandas/__init__.py", line 51, in <module>
    from pandas.core.api import (
  File "/home/pandas-parfaitg/pandas/core/api.py", line 48, in <module>
    from pandas.core.groupby import (
  File "/home/pandas-parfaitg/pandas/core/groupby/__init__.py", line 1, in <module>
    from pandas.core.groupby.generic import (
  File "/home/pandas-parfaitg/pandas/core/groupby/generic.py", line 95, in <module>
    from pandas.core.frame import DataFrame
  File "/home/pandas-parfaitg/pandas/core/frame.py", line 451, in <module>
    class DataFrame(NDFrame, OpsMixin):
  File "/home/pandas-parfaitg/pandas/core/frame.py", line 2643, in DataFrame
    def to_xml(
  File "/home/pandas-parfaitg/pandas/util/_decorators.py", line 387, in decorator
    [
  File "/home/pandas-parfaitg/pandas/util/_decorators.py", line 388, in <listcomp>
    component.format(**params)
KeyError: '""'
```

Reply:
think you have to escape all of the `{` and `}`

Reply:
That was it! (facepalm)